### PR TITLE
Allow resetting celestial vault elements by leaving its arguments empty

### LINKED ISSUE #11916: Leave set_sky() etc. with no arguments to reset a player's celestial vault
## Problem

There is no way to reset a player's celestial vault, if not by storing their previous vault into a temporary table to then reapply it when needed

## Solutions

Have functions like `set_sky()`, `set_sun()` etc. accept no arguments to reset it to default

## Additional context

Usecases:
- events where the sky changes its colour for a temporary amount of time (i.e. literal server events, but also consuming an item, obtaining a power up etc)
- a formspec to test the settings of the celestial vault for a minigame, which it turns to server's default once closed

### PR DESCRIPTION
**Pull request by Zughy**
- How does the PR work? `set_sky()` etc, when left blank, reset that part of the celestial vault to its default
- Does it resolve any reported issue? Fixes #11916
- Does this relate to a goal in [the roadmap](../doc/direction.md)? 2.4, API convenience
- How about the dummy image error it generates? Not this PR's fault, see #11930 

## To do

This PR is Ready for Review.

## How to test

```lua
minetest.register_on_joinplayer(function(player)
    
    local sky = {
        sky_color = {
            day_sky = "#ffffff"    
        }
    }
    
    local sun = {
        texture = "cdb_queued.png",
        size = 2
    }
    
    local moon = {
        texture = "bubble.png",
        size = 2
    }
    
    local stars = {
        count = 3000,
        scale = 0.5
    }
    
    local clouds = {
        thickness = 50,
        height = 30,
        speed = { x = 0.1, z = 5 }
    }
    
    player:set_sky(sky)
    player:set_sun(sun)
    player:set_moon(moon)
    player:set_stars(stars)
    player:set_clouds(clouds)
    
    minetest.after(3, function()
        player:set_sky()
        player:set_sun()
        player:set_moon()
        player:set_stars()
        player:set_clouds()
    end)
end)
```

### CODE REVIEW

**Comment by Zughy** [comment]
Done. Reviewing commit by commit should make your life easier. Also, I got rid of `cloudparams.h` because it was only a file with a struct that could have been declared in `skyparams.h`

**Comment by Zughy** [comment]
This PR must be changed, but it needs what has been highlighted in #11930 first

**Review by rubenwardy — APPROVED** [review]

File: src/skyparams.h [hunk 1]
```diff
@@ -118,4 +141,16 @@ class SkyboxDefaults
 		stars.scale = 1;
 		return stars;
 	}
+
+	const CloudParams getCloudDefaults()
```
Comment (SmallJoker) [inline src/skyparams.h:145]: Idea: Make these functions static. They do not need access to an initialized `SkyboxDefaults` instance.

**Comment by Zughy** [comment]
@SmallJoker done

**Review by SmallJoker — APPROVED** [review]
Looks good.

**Comment by sfan5** [comment]
> In order to work 100%, this PR needs what has been highlighted in #11930 first.

^ Is this the case now or ...?

**Comment by Zughy** [comment]
@sfan5 poor wording, my bad. I meant that the snippet throws the errors I pointed in the other issue, but it's not something related to this PR

File: src/script/lua_api/l_object.cpp [hunk 2]
```diff
@@ -1724,7 +1724,11 @@ int ObjectRef::l_set_sky(lua_State *L)
 	SkyboxParams sky_params = player->getSkyParams();
 	bool is_colorspec = is_color_table(L, 2);
 
-	if (lua_istable(L, 2) && !is_colorspec) {
+	// reset if empty
+	if (lua_isnoneornil(L, 2) && lua_isnone(L, 3))
+		sky_params = SkyboxDefaults::getSkyDefaults();
```
Comment (sfan5) [inline src/script/lua_api/l_object.cpp:1729]: `if` braces must be consistent, either `if (...) { ... } else { ... }` or `if (...) ... else ...` but not `if (...) ... else { ... }`

File: src/remoteplayer.cpp [hunk 3]
```diff
@@ -71,13 +71,7 @@ RemotePlayer::RemotePlayer(const char *name, IItemDefManager *idef):
...
 	SkyboxDefaults sky_defaults;
 
-	m_skybox_params.sky_color = sky_defaults.getSkyColorDefaults();
-	m_skybox_params.type = "regular";
-	m_skybox_params.clouds = true;
-	m_skybox_params.fog_sun_tint = video::SColor(255, 244, 125, 29);
-	m_skybox_params.fog_moon_tint = video::SColorf(0.5, 0.6, 0.8, 1).toSColor();
-	m_skybox_params.fog_tint_type = "default";
-
+	m_skybox_params = sky_defaults.getSkyDefaults();
```
Comment (sfan5) [inline src/remoteplayer.cpp:74]: All of these can be `SkyboxDefaults::get*Defaults()` too

**Comment by Zughy** [comment]
@sfan5 done

**Comment by sfan5** [comment]
You accidentally committed `client/mod_storage.sqlite`, please remove.

**Comment by Zughy** [comment]
Wait what how-
@sfan5 done

**Comment by Zughy** [comment]
@sfan5 the PR still stays "action / change needed", I guess you misclicked?

**Comment by sfan5** [comment]
I'm suggesting you put the fix from #11930 into this PR.

**Comment by Zughy** [comment]
That's not related to this PR nor I know where to put my hands (I tried last week). It's a free PR that does what's intended already :\

**Comment by sfan5** [comment]
I literally provided a diff of the fix, but okay, I did it myself.

**Comment by Zughy** [comment]
...I didn't see the notification about the other issue, I'm sorry

**Comment by Zughy** [comment]
Wait, so this won't cause any problems with texture packs shipping `sun.png` and `moon.png` by default such as RPG16? https://gitlab.com/Df458/rpg16/-/tree/master/misc

**Comment by sfan5** [comment]
It will. The reason is that at startup the sky picks sun.png *if it exists* and else uses the mesh sun. If you ever change the texture it is impossible to return to this state.
Whatever, I'll fix it in a separate PR.

**Pull request merged**